txnsync/bitmask.go

```diff
@@ -21,26 +21,41 @@ import (
 	"errors"
 )
 
+var errIndexOutOfBounds = errors.New("invalid bitmask: index out of bounds")
+var errIndexNotFound = errors.New("invalid bitmask: index not found")
+var errInvalidBitmaskType = errors.New("invalid bitmask type")
+
 //msgp:allocbound bitmask maxBitmaskSize
 type bitmask []byte
 
 // assumed to be in mode 0, sets bit at index to 1
-func (b *bitmask) SetBit(index int) {
+func (b *bitmask) setBit(index int) {
 	byteIndex := index/8 + 1
 	(*b)[byteIndex] |= 1 << (index % 8)
 }
 
-func (b *bitmask) EntryExists(index int, entries int) bool {
+// entryExists converts the bitmask to type 0 (if not already)
+func (b *bitmask) entryExists(index int, entries int) bool {
 	if len(*b) == 0 {
 		return false
 	}
 	if (*b)[0] != 0 {
-		b.expandBitmask(entries)
+		err := b.expandBitmask(entries)
+		if err != nil {
+			return false
```

Comment:
if we've detected this issue, we shouldn't keep it for ourselves. we need to return it upstream to the caller and fail the decoding of this message.